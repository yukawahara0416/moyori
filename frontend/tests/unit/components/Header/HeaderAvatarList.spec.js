import { shallowMount, createLocalVue } from '@vue/test-utils'
import Component from '@/components/Header/HeaderAvatarList.vue'

const localVue = createLocalVue()

let wrapper
let propsData

beforeEach(() => {
  propsData = {
    currentUser: { data: { id: 1, name: 'test' } }
  }

  wrapper = shallowMount(Component, {
    propsData
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('props', () => {
  it('currentUser', () => {
    expect(wrapper.props().currentUser).toStrictEqual(propsData.currentUser)
    expect(wrapper.props().currentUser instanceof Object).toBe(true)
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
