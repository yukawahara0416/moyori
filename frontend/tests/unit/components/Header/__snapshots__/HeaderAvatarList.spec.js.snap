// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<v-list-stub
  tag="div"
  tile="true"
>
  <v-list-item-stub
    activeclass=""
    tag="div"
  >
     さん
  </v-list-item-stub>
   
  <header-avatar-list-mypage-stub
    currentuser="[object Object]"
  />
   
  <header-avatar-list-signout-stub />
</v-list-stub>
`;
